src/main/java/org/broadinstitute/hellbender/tools/walkers/rnaseq/ASEReadCounter.java

```diff
@@ -1,254 +1,245 @@
+        if (vc.getNAlleles() == 1 || vc.getAlternateAllele(0).getBases().length == 0)
             throw new UserException("The file of variant sites must contain heterozygous sites and cannot be a GVCF file containing <NON_REF> alleles.");
 
-        final char altAllele = (char)vc.getAlternateAllele(0).getBases()[0];
+        final char altAllele = (char) vc.getAlternateAllele(0).getBases()[0];
 
         final String siteID = vc.getID();
-        final ReadBackedPileup pileup = filterPileup(context.getBasePileup(), countType, includeReadsWithDeletionAtLoci());
+        final ReadPileup pileup = filterPileup(alignmentContext.getBasePileup(), countType);
 
         // count up the depths of all and QC+ bases
-        return calculateLineForSite(pileup, contig, position, siteID, refAllele, altAllele);
+        calculateLineForSite(pileup, contig, position, siteID, refAllele, altAllele);
+    }
 
+    @Override
+    public void closeTool() {
+        if (outputStream != null)
+            outputStream.close();
     }
 
-    protected ReadBackedPileup filterPileup(final ReadBackedPileup originalPileup, final CoverageUtils.CountPileupType countType, final boolean includeDeletions){
+    protected ReadPileup filterPileup(final ReadPileup originalPileup, final CountPileupType countType) {
 
-        ReadBackedPileup pileupWithDeletions;
-        if(countType.equals(CoverageUtils.CountPileupType.COUNT_FRAGMENTS_REQUIRE_SAME_BASE))
-            pileupWithDeletions = originalPileup.getOverlappingFragmentFilteredPileup(true,true);
-        else if(countType.equals(CoverageUtils.CountPileupType.COUNT_READS))
+        ReadPileup pileupWithDeletions;
+        if (countType.equals(CountPileupType.COUNT_FRAGMENTS_REQUIRE_SAME_BASE))
+            pileupWithDeletions = originalPileup.getOverlappingFragmentFilteredPileup(true, true);
+        else if (countType.equals(CountPileupType.COUNT_READS))
             pileupWithDeletions = originalPileup;
-        else if(countType.equals(CoverageUtils.CountPileupType.COUNT_FRAGMENTS))
-            pileupWithDeletions = originalPileup.getOverlappingFragmentFilteredPileup(false,true);
+        else if (countType.equals(CountPileupType.COUNT_FRAGMENTS))
+            pileupWithDeletions = originalPileup.getOverlappingFragmentFilteredPileup(false, true);
         else
             throw new UserException("Must use valid CountPileupType");
 
-        return includeDeletions ? pileupWithDeletions: pileupWithDeletions.getPileupWithoutDeletions();
+        return pileupWithDeletions.makeFilteredPileup(p -> !p.isDeletion());
 
     }
 
-    protected String calculateLineForSite(final ReadBackedPileup pileup, final String contig, final long position, final String siteID, final char refAllele, final char altAllele){
+    protected void calculateLineForSite(final ReadPileup pileup, final String contig, final long position, final String siteID, final char refAllele, final char altAllele) {
```

Comment:
Because there's one case that needs to not output anything (including a new line) I'm having `calculateLineForSite` return `null` in that case. Do you have a cleaner solution than having the function be void and output the line or return a `StringBuilder` that's null in that one case?